features/step_definitions/networking.rb

```diff
@@ -790,3 +790,103 @@
   @result = _admin.cli_exec(:get, resource: "network.operator", output: "jsonpath={.items[*].spec.defaultNetwork.type}")
   raise "the networkType is not #{network_type}" unless @result[:response] == network_type
 end
+
+Given /^the bridge interface named "([^"]*)" with address "([^"]*)" is added to the "([^"]*)" node$/ do |bridge_name,address,node_name|
+  ensure_admin_tagged
+  node = node(node_name)
+  host = node.host
+  @result = host.exec_admin("ip link add #{bridge_name} type bridge;ip address add #{address} dev #{bridge_name};ip link set up #{bridge_name}")
+  raise "Failed to add  bridge interface" unless @result[:success]
+end
+
+Given /^a DHCP service is configured for interface "([^"]*)" on "([^"]*)" node with address range and lease time as "([^"]*)"$/ do |br_inf,node_name,add_lease|
+  ensure_admin_tagged
+  node = node(node_name)
+  host = node.host
+  #Following will take dnsmasq backup and append curl contents to the dnsmasq config after
+  @result = host.exec_admin("cp /etc/dnsmasq.conf /etc/dnsmasq.conf.bak;curl https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/dnsmasq_for_testbridge.conf | sed s/testbr1/#{br_inf}/g | sed s/88.8.8.100,88.8.8.110,24h/#{add_lease}/g >> /etc/dnsmasq.conf;systemctl restart dnsmasq --now")
+  raise "Failed to configure DNS server" unless @result[:success]
+  step "10 seconds have passed"
```

Comment:
@anuragthehatter perhaps we can loop with a timeout to check for the DNS server status instead of the hardcoded wait time?

Reply:
@pruan-rht Yep, thats a great suggestion. Let me check on that.